Dockerfile

```diff
@@ -0,0 +1,19 @@
+# Note: FROM command is inserted by Makefile
+# TODO: As of Docker-CE 17.05, we could use this better approach instead:
+#ARG base_layer
+#FROM $base_layer
+
+MAINTAINER Ben Barsdell <benbarsdell@gmail.com>
+
+ARG make_args
+
+# Build the library
+WORKDIR /bifrost
+COPY . .
```

Comment:
I find it can be useful to have the git history inside the container, but perhaps I should be doing that with your approach of using the base container and mapping the source directory instead. In that case, this sounds like a good idea.